tests/pass/shims/windows-fs.rs

```diff
@@ -0,0 +1,131 @@
+//@only-target: windows # this directly tests windows-only functions
+//@compile-flags: -Zmiri-disable-isolation
+#![allow(nonstandard_style)]
+
+use std::os::windows::ffi::OsStrExt;
+use std::ptr;
+
+#[path = "../../utils/mod.rs"]
+mod utils;
+
+// Windows API definitions.
+type HANDLE = isize;
+type BOOL = i32;
+type DWORD = u32;
+type LPCWSTR = *const u16;
+
+const GENERIC_READ: u32 = 2147483648u32;
+const GENERIC_WRITE: u32 = 1073741824u32;
+pub const FILE_SHARE_NONE: u32 = 0u32;
+pub const FILE_SHARE_READ: u32 = 1u32;
+pub const FILE_SHARE_WRITE: u32 = 2u32;
+pub const OPEN_ALWAYS: u32 = 4u32;
+pub const OPEN_EXISTING: u32 = 3u32;
+pub const CREATE_NEW: u32 = 1u32;
+pub const FILE_ATTRIBUTE_DIRECTORY: u32 = 16u32;
+pub const FILE_ATTRIBUTE_NORMAL: u32 = 128u32;
+pub const FILE_FLAG_BACKUP_SEMANTICS: u32 = 0x02000000u32;
+
+#[repr(C)]
+struct FILETIME {
+    dwLowDateTime: DWORD,
+    dwHighDateTime: DWORD,
+}
+
+#[repr(C)]
+struct BY_HANDLE_FILE_INFORMATION {
+    dwFileAttributes: DWORD,
+    ftCreationTime: FILETIME,
+    ftLastAccessTime: FILETIME,
+    ftLastWriteTime: FILETIME,
+    dwVolumeSerialNumber: DWORD,
+    nFileSizeHigh: DWORD,
+    nFileSizeLow: DWORD,
+    nNumberOfLinks: DWORD,
+    nFileIndexHigh: DWORD,
+    nFileIndexLow: DWORD,
+}
+
+#[repr(C)]
+struct SECURITY_ATTRIBUTES {
+    nLength: DWORD,
+    lpSecurityDescriptor: *mut std::ffi::c_void,
+    bInheritHandle: BOOL,
+}
+
+extern "system" {
+    fn CreateFileW(
+        file_name: LPCWSTR,
+        dwDesiredAccess: DWORD,
+        dwShareMode: DWORD,
+        lpSecurityAttributes: *mut SECURITY_ATTRIBUTES,
+        dwCreationDisposition: DWORD,
+        dwFlagsAndAttributes: DWORD,
+        hTemplateFile: HANDLE,
+    ) -> HANDLE;
+    fn GetFileInformationByHandle(
+        handle: HANDLE,
+        file_info: *mut BY_HANDLE_FILE_INFORMATION,
+    ) -> BOOL;
+    fn CloseHandle(handle: HANDLE) -> BOOL;
+    fn GetLastError() -> DWORD;
+}
```

Comment:
Is there some crate we can use to avoid having to carry all these definitions ourselves?

`pass-dep` tests can use the dependencies declared in `test_dependencies/Cargo.toml`.

Reply:
Ah, cool. I just followed the other *-windows shim tests, many of these are almost literally copied from `windows_sys`.

(I can add a follow up task to my to-do list for other windows shim tests defining their own API if we want)

Reply:
> Ah, cool. I just followed the other *-windows shim tests, many of these are almost literally copied from windows_sys.

Yeah, fair. But if we're now getting more of these tests IMO we should start doing this the proper way. We also use the libc crate for Unix targets, so using the windows crate here is more consistent as well.